internal/kafka/internal/handlers/data_plane_cluster_test.go

```diff
@@ -0,0 +1,701 @@
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(nil), //&private.DataPlaneClusterUpdateStatusRequest{}, //SampleDataPlaneclusterStatusRequestWithAvailableCapacity(),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: false,
+		},
+		{
+			name: "should return an error with invalid NodeInfo when validating body",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.NodeInfo.Ceiling = nil
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error with invalid ResizeInfo when validating body",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.ResizeInfo.NodeDelta = nil
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error with invalid Total when validating body",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.Total.Connections = nil
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error with invalid Remaining when validating body",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.Remaining.Connections = nil
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error with invalid Strimzi when validating body",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.Strimzi = []private.DataPlaneClusterUpdateStatusRequestStrimzi{
+						{
+							Ready:            true,
+							Version:          "",
+							KafkaVersions:    []string{},
+							KafkaIbpVersions: []string{},
+						},
+					}
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewDataPlaneClusterHandler(tt.fields.dataplaneClusterService)
+			err := h.validateBody(tt.args.request)()
+			Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+
+func Test_validateNodeInfo(t *testing.T) {
```

Comment:
NodeInfo
Total
Remaining
ResizeInfo

are being removed in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1068 we might want to remove the test case for them as they'll be removed anytime soon depending on when the linked PR is merged

Reply:
thanks for your review @machi1990. I will hold off with updating this PR after the #1068 is merged. Otherwise the tests will fail